### DISCUSSION TITLE
Roadblock to be part of road network

### DISCUSSION THREAD

**Discussion by catalin-duta** [Keriew/augustus#526] [Ideas]
Given the below situation:
![Screenshot_20211112-144555152](https://user-images.githubusercontent.com/93785920/141469552-8ef9f2be-69b6-4925-aba6-42ffaeadfda9.jpg)
There is technically a road below the roadblock but it does not count for the connected road requirement.

The road is only there in the first place to provide the road access, because otherwise the granary is only connected to other granaries (and that unfortunately does not count as roads).

In this example:
![Screenshot_20211112-144858257](https://user-images.githubusercontent.com/93785920/141469862-5515d397-f7fa-418b-b5bc-2365672c8cbf.jpg)
I would put roadblocks to the entrance of both granaries to prevent roamers from going astray NW.
But doing that blocks labor access for the granary on the right. 
And putting a roadblock just in front of the granary on the left still lets roamers pass behind the granaries by going through both.

**Comment by Keriew** [comment]
Buildings require a non-roadblock tile to allow them to spawn walkers (at the very least a labour seeker walker). This also allows the usage of roadblocks to decide which tile the buildings will spawn walkers (in case it's connected to multiple separate road systems) and is consistent with how the future games handled it. Because of this, I don't think we'll change it.

**Reply by catalin-duta** [reply]
So even if a labor seeker is allowed through that roadblock, he will not spawn on it, correct?

I haven't tried connecting the same building to multiple road networks, I remember it was buggy in vanilla. Can buildings in Augustus be connected to, and properly working on, multiple road systems?

**Reply by Keriew** [reply]
> So even if a labor seeker is allowed through that roadblock, he will not spawn on it, correct?

That's correct. That's also how it works in _Emperor_. I considered changing that, but never got around to doing it.

Buildings connected to multiple road networks will only act as connected to one of them, same as in original _Caesar 3._

**Reply by catalin-duta** [reply]
Very good to know, thank you!

**Comment by crudelios** [comment]
This was discussed a long way back, then Augustus was still called julius-gc. We settled like this by design.

However, I think roamers will only use the granary's intersection if they're not dead ends, otherwise they won't turn there.

Also, totally unrelated, @catalin-duta, are you playing on an Android phone? What's the experience like?

**Reply by catalin-duta** [reply]
The purpose is to block the roamer from entering the granary at all, because it is not a dead end in all directions.
Very rough sketch:
![Screenshot_20211112-202940720_1](https://user-images.githubusercontent.com/93785920/141517221-fa93f054-5959-43e7-ad96-712efb53bc02.jpg)
- roadblock on blue position works without blocking the granary, because it has a road on the other side.
- roadblock on green position blocks the granary
- without the roadblock on green, the market lady will take the orange path, serving lunch to the quarry workers (hmm)

Yes, I'm playing on an Android phone and the experience is great! I could think of a few small quality of life improvements, but I personally find it even more enjoyable than if I were to be playing on a laptop or a computer. And it definitely beats other games when riding the subway because it doesn't need an internet connection, haha!


The comment above gave me an idea: what if market ladies *did* deliver lunch to workers in workshops or quarries, and gave them some productivity bonus with that?
